xmtp_mls/src/groups/message_history.rs

```diff
@@ -384,58 +592,50 @@ pub(crate) fn decrypt_history_file(
     Ok(())
 }
 
-fn create_bundle_hash(
+async fn upload_history_bundle(
+    url: &str,
     file_path: PathBuf,
-    signing_key: &[u8],
-) -> Result<(Vec<u8>, String), MessageHistoryError> {
+) -> Result<String, MessageHistoryError> {
     let mut file = File::open(file_path)?;
     let mut content = Vec::new();
     file.read_to_end(&mut content)?;
 
-    let key = hmac::Key::new(hmac::HMAC_SHA256, signing_key);
-    let tag = hmac::sign(&key, &content);
-    let hmac_hex = hex::encode(tag.as_ref());
-    Ok((content, hmac_hex))
-}
-
-async fn upload_history_bundle(
-    url: &str,
-    file_path: PathBuf,
-    signing_key: &[u8],
-) -> Result<(), MessageHistoryError> {
-    let (content, hmac_hex) = create_bundle_hash(file_path, signing_key)?;
-
     let client = reqwest::Client::new();
-    let _response = client
-        .post(url)
-        .header("X-HMAC", hmac_hex)
-        .body(content)
-        .send()
-        .await?;
+    let response = client.post(url).body(content).send().await?;
 
-    Ok(())
+    if response.status().is_success() {
+        Ok(response.text().await?)
+    } else {
+        eprintln!(
+            "Failed to upload file. Status code: {} Response: {:?}",
+            response.status(),
+            response
+        );
+        Err(MessageHistoryError::Reqwest(
+            response.error_for_status().unwrap_err(),
```

Comment:
@insipx `clippy` is complaining about this line in CI, but doesn't complain when i run it locally. any suggestions on a solution? how can i get `clippy` to recognize this issue locally?

Reply:
nvm, i see that it's been fixed in `main`

Reply:
ah weird, it should who locally. The only difference for CI is that we add -Dwarnings to fail the build. Maybe you're on an old clippy build, possibly `rustup update stable`?

running just `cargo clippy` will show a warning but not fail the build